Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -1328,6 +1303,91 @@ async def fetch_channels_async():
         await asyncio.sleep(5)
 
 
+async def fetch_users_iterable():
+    cursor = None
+    updated_users_list: list = []
+
+    res = await ASYNC_CLIENT.users_list(limit=PAGINATED_COUNT, cursor=cursor)
+    integration_context = get_integration_context(SYNC_CONTEXT)
+    now = datetime.now()
+    timeout_safety = return_next_interval(fetch_interval=TIMEOUT_SAFETY_NET, last_update_time=now)
+    while True:
+        if now >= timeout_safety:
+            # This provides the function with a safety net to prevent the fetch process itself from taking longer than
+            # 10 minutes. If the workspace has more than 120,000 users or fetching the updated list takes longer than
+            # 10 minutes, we will store the users the function has collected so far and end the fetch.
+            demisto.debug(f"Timeout safety net was exceeded. Current fetch_users_iterable run is being abandoned.\n"
+                          f"The timeout_safety variable is {timeout_safety} and the current time is {now}.")
+            break
+        cursor = res.get('response_metadata', {}).get('next_cursor')
+        members = res['members'] if res and res.get('members') else []
+        demisto.debug(f"Length of members received from Slack api is {len(members)}")
+        for member in members:
+            updated_member: dict = {
+                'name': member.get('name'),
+                'id': member.get('id'),
+                'real_name': member.get('real_name', ''),
+                'profile': {
+                    'email': member.get('email', '')
+                }
+            }
+            updated_users_list.append(updated_member)
+        if not cursor:
+            demisto.info("Finished updating users list")
+            break
+        total_try_time = 0
+        while total_try_time < MAX_LIMIT_TIME:
+            try:
+                res = await ASYNC_CLIENT.users_list(limit=PAGINATED_COUNT, cursor=cursor)
+
+            except SlackApiError as api_error:
+                demisto.debug(f'Got rate limit error (sync). Body is: {str(res)}\n{api_error}.'
+                              f'Retries count is - {total_try_time} out of max retries limit: {MAX_LIMIT_TIME}')
+                response = api_error.response
+                headers = response.headers  # type: ignore
+                if 'Retry-After' in headers:
+                    retry_after = int(headers['Retry-After'])
+                    total_try_time += retry_after
```

Comment:
you should increment total_try_time before if 'Retry-After' in headers because now there is non increment inn case 'Retry-After' not in headers.